### DISCUSSION TITLE
Why use unix domain sockets？

### DISCUSSION THREAD

**Discussion by Lxiaoyouling** [cloud-hypervisor/cloud-hypervisor#3951] [Q&A]
Hello,

In live migration, I notice that cloud-hypervisor communicates through Unix Domain sockets, which is inconvenient for two host machines to communicate. I wonder why use unix domain sockets?

Looking forward to your reply!

**Comment by sboeuf** [comment]
You can simply use socat to perform the redirection, as mentioned at the end of this document https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/live_migration.md

**Comment by rbradford** [comment]
I assume you're talking in the context of migration. This is so that the orchestration infrastructure can implement its own transport. e.g. by applying some authentication and using TLS for its transport. If CH had to implement the handling of IP and TLS and authentication it would bloat the project.

**Comment by Lxiaoyouling** [comment]
But for libvirt's existing migration interface, like

virsh migrate --live ... qemu+tcp://10.111.7.132/system

If use unix sockets, I need to start a cloud-hypervisor process on the target host and then use socat to connect the unix socket to the TCP port. This is inconsistent with libvirt's existing interface, which is not very convenient.

**Reply by sboeuf** [reply]
To solve that kind of issue, I guess you would need to update the libvirt CH driver.

**Comment by Lxiaoyouling** [comment]
I simply modified the code and replaced UnixStream with TcpStream, which can realize tcp transmission. But I need to start cloud-hypervisor with --seccomp false. Does seccomp affect tcp?

**Reply by sboeuf** [reply]
Oh yes you can definitely use `--seccomp false`, that won't affect TCP.
